### PR TITLE
[buddy] fix(helm-chart): adjust to entrypoint within dockerfile

### PR DESCRIPTION
**Pull request by fheinecke**
Signed-off-by: Fred Heinecke <fred.heinecke@goteleport.com>

changelog: Fixed event-handler Helm charts using the wrong command when starting the event-handler container

### CODE REVIEW

**Review by zmb3 — APPROVED** [review]

**Review by r0mant — APPROVED** [review]

**Comment by hugoShaka** [comment]
Can we also add in the v16 changelog that we did a breaking change by renaming the binary?

**Comment by fheinecke** [comment]
> a breaking change by renaming the binary

IMO this is not a breaking change as binary names in container images aren't considered an intentional part of our user-facing interface.

**Comment by hugoShaka** [comment]
> IMO this is not a breaking change

We even broke our own chart.

**Comment by hugoShaka** [comment]
Many deployments rely on custom entrypoints and init scripts to obtain credentials for the plugin ( mostlypre-machineID setups). I would rather revert the change as there was no need to break by renaming binaries in the container. This will generate support load.

**Review by hugoShaka — CHANGES_REQUESTED** [review]
Many deployments rely on custom entrypoints and init scripts to obtain credentials for the plugin (mostly pre-machineID setups). I would rather revert the change or add a symlink as there was no need to break by renaming binaries in the container. This will generate support load.

**Comment by fheinecke** [comment]
> We even broke our own chart.

Yes - but again, this is not intended as a public, customer-facing interface. This is equivocal to changing a variable name in our Teleport Go code and then claiming that it's a breaking customer change for customers when we miss a reference to it somewhere in our codebase. A customer could be building their own Teleport builds with their own patches that depend on this variable (just as they could be building their own helm charts/deployment manifests), and a change such as this would break their workflow, but that doesn't mean that we support it in the first place.

> This is a breaking change as this will cause user's deployment to break

Will this cause deployments based on our Helm chart to break (outside of the bug that this PR fixes), or just custom deployment manifests that users' may be using?

> make the v15 chart not able to deploy a v16 handler.

What is the use case for this? AFAIK the auto updater doesn't support plugins, and we version our Helm charts with Teleport.

---

Given that this is the first time we've heard of the issue in the month+ that it's been released, and that it makes our build process easier and simpler, I am not inclined to revert this change.

**Comment by hugoShaka** [comment]
> Yes - but again, this is not intended as a public, customer-facing interface.

Regardless of the intent; Customers, and our own solution engineers are invoking the binary by its name in their setups. This change is disruptive for all those setups and I don't understand what is the motivation to do a change which will generate support load in this case.

**Comment by fheinecke** [comment]
> our own solution engineers are invoking the binary by its name in their setups

Why? What is the use case? To be honest even with our Helm charts I don't understand why we're invoking it via `command` rather than just passing `args`.

> what is the motivation to do a change which will generate support load in this case

Quite literally every change to our repos which customers can see or use could affect customers in some way. The questions we should ask are:
* How many customers will this affect?
* How impactful is the change on our customers?
* Does this change affect a customer-facing interface that we provide support for?
* How impactful is the change on us (on things like support load, or our internal processes)?

To answer these questions for this change:
* How many customers will this affect? - All customers using the event handler container image that also explicitly invoke the binary inside the container image by name/path. Given that this is the first report across v15 and v16 that we've seen, there is likely either not many customers impacted, or other impact customers have fixed the issue themselves.
* How impactful is the change on our affected customers? Not terribly large - customers simply need to change `<plugin name>` e.g. `event-handler` to `teleport-plugin`. 
* Does this change affect a customer-facing interface that we provide support for? No - the paths of binaries inside container images is not something that we intend on being stable.
* How impactful is the change on us (on things like support load, or our internal processes)? Given that this is the first report of the issue across all plugins and two major versions over the past 6 weeks, and that this means less complexity and maintenance for our build process, the gain so far has been a net positive for us as a company.

In summary, this change to an internal interface has a small impact on a very small number of customers, and it brings us a net positive gain. As such, I believe the benefits to the company as a whole outweigh the cost.
